### ISSUE TITLE
Finish the OpenAPI integration

### ISSUE THREAD

**Issue by arteymix** [PavlidisLab/Gemma#174]
Before we make it available on the production instance, there's still a couple of things to address about the OpenAPI integration

 - [x] add a mapping for `/resources/restapidocs -> /resources/restapidocs/index.html`
 - [x] fix parameter types derived from model annotations (see https://github.com/swagger-api/swagger-core/issues/3992)
 - [ ] migrate the theme for the current REST API docs to Swagger UI
 - [ ] port all the docs written for the current REST API
 - [x] declare authentication scheme
 - [ ] declare error responses and systematically annotate the endpoints (maybe via explicit `throws` statement even for runtime exception, or [@ApiResponse](https://docs.swagger.io/swagger-core/v2.1.10/apidocs/io/swagger/v3/oas/annotations/responses/ApiResponse.html))
 - [ ] integrate code snippets (@yavyx would be of great help here with the R client)
 - [ ] #224
 - [x] ~~remove the `ValueObject` suffix to the model definitions, the frontend does not need to know that detail~~
 - [x] use ISO 8601 to format dates #301 
 - [x] fix parameters using inheritance (i.e. `TaxonArg` `->` `TaxonIdArg` | `TaxonNcbiIdArg` | `TaxonStringArg`)
 - [x] https://github.com/PavlidisLab/Gemma/issues/311
 - [ ] https://github.com/PavlidisLab/Infrastructure/issues/21
 - [ ] https://github.com/PavlidisLab/Infrastructure/issues/22 (only once released!)
 - [x] hide or properly reference `IdentifiableValueObjectIdentifiable` (should be simply `IdentifiableValueObject`)

For the `index.html` issue, I expected the `<welcome-file-list/>` to handle this case since the delivery is delegated to the default servlet (i.e. handled by Tomcat itself), but it just doesn't work.

For parameter types, the alternative of annotating types at model-level is absolutely not something we want to do: we would have to annotate each individual mentions. Just to clarify, we want this:

```java
class IntArg {
    private Integer value;
}
```

which is serialized as `{type: "object", properties: {"value": {"type": "integer"}}}` to be annotated simply as: `{"type": "integer"}`.

**Comment by arteymix** [comment]
I think we can remove the standalone preset, unless there's a good reason to keep it.

Also, for the cookie-based authentication, it would be nice to just use the `JSESSIONID` without asking for it in the UI. I think there's a mechanism in the config to set it up.

**Comment by arteymix** [comment]
The standalone preset was removed in 6de71effe050dc6e7e2fce4dfd8f1dbcaea69cb9

**Comment by arteymix** [comment]
I fixed the majority of the cases using a custom `ModelConverter` to recreate the type annotations from the class definition. It definitely sounds like a bug in Swagger.

**Comment by arteymix** [comment]
There's still issues with array of parameters (from `AbstractArrayArg`) and inheritance (i.e. `TaxonArg` -> `TaxonIdArg`, `TaxonStringArg`, etc.).

**Comment by arteymix** [comment]
The problem is not with the array of entity, but rather the entity inheritance itself. The schema of `TaxonArg` for example is correct, but it has extraneous `type` and `properties` attributes that are unnecessary:

```json
{"type":"object","properties":{"value":{"type":"object"}},"oneOf":[{"type":"integer","description":"A numerical taxon identifier.","format":"int64"},{"type":"integer","description":"An NCBI taxon identifier.","format":"int64","externalDocs":{"url":"https://www.ncbi.nlm.nih.gov/taxonomy"}},{"type":"string","description":"A taxon identifier that matches either its scientific or common name."}]}
```

**Comment by arteymix** [comment]
Parameter inheritance has been fixed! We're stripping type and properties attributes on annotated abstract classes to deal with that.

**Comment by arteymix** [comment]
GEEQ scores fields should be renamed, the getter is in the name:

![image](https://user-images.githubusercontent.com/1318477/157745667-9c52bbff-40da-40d0-ba12-9a38543d8305.png)

**Comment by arteymix** [comment]
I fixed the redirection + rewrite in 80cf6856d48074fe060fd2a53e0d585ee7ada6e5

**Comment by arteymix** [comment]
Essentially, the welcome file list only applies to the root of the website and not subdirectories.

**Comment by arteymix** [comment]
There's a mysterious `IdentifiableValueObjectIdentifiable` appearing in the list of models:

![image](https://user-images.githubusercontent.com/1318477/192394038-8e100b7a-cb82-49ec-a46d-9c63497d5010.png)

**Comment by arteymix** [comment]
Ok, this was caused by `SearchResultValueObject` reference to the type. It's fixed now.

**Comment by arteymix** [comment]
I'll be closing this. All remaining issues are either infrastructure-related or postponed to the next release.
